Разметка внутренней страницы

fesspav opened the pull request:
Добавил html-разметку главной страницы, создал папку images и скачал туда картинки, создал css-файл и попытался подключить шрифты

---
:mortar_board: [Разметка главной страницы](https://up.htmlacademy.ru/htmlcss/33/user/1901121/tasks/3)

---
:mortar_board: [Разметка внутренней страницы](https://up.htmlacademy.ru/htmlcss/33/user/1901121/tasks/4)

fesspav committed:
Added html-markup

Добавил html-разметку главной страницы, создал папку images и скачал туда картинки, создал css-файл и попытался подключить шрифты

fesspav committed:
Добавил nude

fesspav committed:
Поменял разметку к логотипу в header

Добавил точки в alt (согласно гайда)

fesspav committed:
Добавил страницу каталога

efiand commented on blank.html:
```diff
@@ -10,9 +10,48 @@
-        <a href="#" class="navigation-logo">
-          <img src="" alt="Логотип города Седона" class="page-header-logo" />
-        </a>
+        <svg
```
Все-таки `<img>` лучше, так как:
* Грузится один раз, далее кэшируется
* Дает пользователю возможность взаимодействовать в контекстном меню именно как с изображением

Поэтому делаем так:
если можем - добавляем через img. Не можем, только когда картинка должна менять цвет при интерактивном взаимодействии (именно цвет, а не прозрачность).

efiand commented on blank.html:
```diff
@@ -10,9 +10,48 @@
+        </svg>
         <ul class="navigation-list">
           <li class="navigation-item">
             <a href="#" class="navigation-link">Главная</a>
```
Ссылка очевидна: index.html

efiand commented on blank.html:
```diff
@@ -28,16 +67,16 @@
           <li class="navigation-item">
             <a href="" class="navigation-link">
-              <span class="visually-hidden">Лайки</span>
+              <span class="visually-hidden">Лайки.</span>
```
Полагаю, что это все-таки Избранное текущего пользователя и количество элементов в нём.

https://up.htmlacademy.ru/htmlcss/33/criteries#TEST-06
(критерий про соблюдение требований макета).
На макете указано количество добавленных в избранное материалов (ну или лайков, если на то пошло). В разметке их нет.

efiand commented on blank.html:
```diff
@@ -28,16 +67,16 @@
           </li>
           <li class="navigation-item">
-            <a href="#" class="navigation-link-button button">Хочу сюда!</a>
+            <button type="navigation-link button">Хочу сюда!</button>
```
Что делает эта кнопка на ЭТОЙ странице? Была бы ссылка - было бы понятно, что на нее потом адрес добавят.

efiand commented on blank.html:
```diff
@@ -53,33 +92,33 @@ <h1 class="main-title">404 ошибка</h1>
         <ul class="footer-social-list">
           <li class="footer-social-item">
             <a href="https://twitter.com/htmlacademy_ru" class="button">
```
Обычно класс `button` используется для стилизации кнопок и ссылок "типичного" кнопочного вида - с фоном и паддингами. Эти ссылки вообще ни разу не похожи на типичные кнопки, у них по идее должен быть свой класс из пространства имен `footer-social-`

efiand commented on blank.html:
```diff
@@ -53,33 +92,33 @@ <h1 class="main-title">404 ошибка</h1>
+        <h2 class="visually-hidden">Логотип автора сайта.</h2>
         <a href="https://htmlacademy.ru" target="_blank">
-          <img src="" alt="HTML Академия" class="footer-logo" />
+          <img src="" alt="HTML Академия." class="footer-logo" />
```
Компания носит немного иное название.

efiand commented on blank.html:
```diff
@@ -53,33 +92,33 @@ <h1 class="main-title">404 ошибка</h1>
-        <h2 class="visually-hidden">Телефон для связи</h2>
+        <h2 class="visually-hidden">Телефон для связи.</h2>
         <address class="footer-contacts-address">
           <p class="footer-cotacts-address-phone">+7(495) 666-02-66</p>
```
https://up.htmlacademy.ru/htmlcss/33/criteries#HTML-03 - телефон должен быть ссылкой

efiand commented on source/index.html:
```diff
@@ -6,7 +6,7 @@
 
```
Такое количество пустых строк нежелательно.

efiand commented on source/index.html:
```diff
@@ -25,24 +25,24 @@ <h2>Настоящий <br> городок</h2>
                 </li>
                 <li class="adventages-item">
-                  <img src="" alt="Жилье" class="icon">
+                  <img src="" alt="" class="icon">
```
https://up.htmlacademy.ru/htmlcss/33/criteries#IMG-01

Это не контентные изображения, потому что они не показывают нам реальное жильё Седоны. Это декорации, жестко привязанные к дизайн-системе этого сайта. В остальных пунктах этого списка - то же самое.

efiand commented on source/index.html:
```diff
@@ -25,24 +25,24 @@ <h2>Настоящий <br> городок</h2>
 
               <ul class="advantages-item first-list">
                 <li class="advantages-item advantages-item-pic">
```
Эта картинка неоднородна по отношению к трем другим элементам данного вложенного списка (в огороде бузина, а в Киеве дядька). Те называют конкретные моменты и дают некоторое описание, а этот пункт - просто показывает общую картину на тему первой причины (по аналогии, как Мост дьявола - во второй причине).

efiand commented on source/index.html:
```diff
@@ -51,7 +51,7 @@ <h2 class="advantage-title">Сувениры</h2>
           </li>
           <li class="advantages-item picture">
-            <img src="" alt="Окрестные пейзажи">
+            <img src="" alt="Окрестные пейзажи.">
```
Это и есть изображение Моста дьявола. Не надо было отрывать картинку от соответствующего списка.

efiand commented on source/index.html:
```diff
@@ -51,7 +51,7 @@ <h2 class="advantage-title">Сувениры</h2>
+            <img src="" alt="Окрестные пейзажи.">
           </li>
           <li class="advantages-item second">
             <h2>Там есть мост дьявола</h2>
```
Классы с названием firth и т. п. тут плохи, так как при изменении количества причин или добавлению картинки к одной из них список разрастется, и нужно воспринимать элементы сеток не как "третий", "четвертый", "пятый", а как "первый в ряду", "второй в ряду", "третий в ряду и т.п.". Свойства ячеек одинакового размера обычно не отличаются, и давать уникальные классы одинаковым ячейкам не нужно.

efiand replied:
Классы `button` и `btn` на одном проекте - нехорошая практика, они обозначают одно и то же (некоторую универсальную типовую кнопку).

Одним и тем же классом `button` у тебя обозначены совершенно разные по внешнему виду сущности. Это странно. Одинаковые с виду вещи делаются одинаковыми классами, разные - разными.

efiand commented on source/index.html:
```diff
@@ -0,0 +1,108 @@
+
+      <div class="services">
+        <div class="service find-hotel">
+          <h2 class="find-hotel-title">Заинтересовались?</h2>
```
Рекомендую обратиться к учебнику, заголовки должны быть информативными, отражать то, что написано ниже. Тут скрытый заголовок примерно такой: "Поиск гостиницы"
А данный текст - это параграф или тег стилистического выделения.

Да, стилизуется он тем же классом, что стоит на некоторых заголовках, но семантика это одно, а стилизация это другое, не нужно их пересекать.

efiand commented on source/index.html:
```diff
@@ -0,0 +1,108 @@
+          <a href="" class="find-hotel-btn btn">ПОИСК ГОСТИНИЦЫ В седоне</a>
+        </div>
+        <div class="service subscribe">
+          <h2 class="email-subscribe-title">Подпишитесь на рассылку</h2>
```
Одинаковые вещи делаются одинаково. Тут заголовок такой же, как слово "Заинтересовались" выше, и текст под ним тоже. И заголовок с текстом над списком причин - тоже. Зачем разные классы, какую задачу решают? Только усложняют задачу стилизации, заставляя их все перечислять в CSS, и ничего не дают ни роботу, ни пользователям, ни разработчикам (им как раз понимание, что это одинаковая стилистическая сущность, помогла бы решить проблему заказчика "давайте мы для заголовков увеличим шрифт")

efiand commented on source/index.html:
```diff
@@ -0,0 +1,108 @@
+          <h2 class="email-subscribe-title">Подпишитесь на рассылку</h2>
+          <p class="email-subscribe-description">Только полезная информация и никакого спама, <br>
+            честное бойскаутское!</p>
+          <form class="subscribe-form" action="https://echo.htmlacademy.ru/" method="post" enctype="multipart/form-data">
```
Зачем менять enctype, если тут нет файлов?

efiand commented on source/catalog.html:
```diff
@@ -0,0 +1,207 @@
+        <legend class="catalog-filter-title">Стоимость, ₽:</legend>
+          <label class="min-price"><input type="number" name="min-price" value="0"> от</label>
+          <label class="max-price"><input type="number" name="max-price" value="9000"> до</label>
+          <input class="slider" type="range" min="0" max="12000" value="9000">
```
Пусть пока так, потом покажут на лайвах

efiand commented on source/catalog.html:
```diff
@@ -0,0 +1,207 @@
+          <input class="slider" type="range" min="0" max="12000" value="9000">
+      </fieldset>
+
+    <button class="button" type="submit">Применить</button>
```
https://up.htmlacademy.ru/htmlcss/33/criteries#PROJ-01
Единообразный стиль предполагает, что отступы выдерживаются.

efiand commented on source/catalog.html:
```diff
@@ -0,0 +1,207 @@
+      </fieldset>
+
+    <button class="button" type="submit">Применить</button>
+    <input class="button-reset" type="reset" value="Сбросить">
```
лучше использовать `<button>`, у него больше возможностей для стилизации. Везде одинаковый подход лучше, а button всегда лучше input в этом случае. input этих типов морально устарели

efiand commented on source/catalog.html:
```diff
@@ -0,0 +1,207 @@
+      </li>
+    </ul>
+
+    <hr>
```
https://up.htmlacademy.ru/htmlcss/33/criteries#HTML-06
Это устаревший тег, который не имеет семантического значения. Бордеры делаем стилями.

fesspav replied:
удалил

efiand commented on source/catalog.html:
```diff
@@ -0,0 +1,207 @@
+    </ul>
+
+    <div class="count-pages">
+      <label>Гостиниц на странице:
```
https://up.htmlacademy.ru/htmlcss/33/criteries#HTML-05
Все поля обернуты в формы. Этот селект служит не для сортировки внутри одной страницы, он служит для подгрузки данных, следовательно, это форма.

fesspav replied:
не понял что делать.
я предполагаю, что на странице установлено значение вывода результатов 10 по умолчанию, если количество результатов меньше, в ячейке пишется текущее количество, если больше, то 10, а дальше можно подгрузить еще (20, 50, 100)

efiand replied:
Извини, я не на тот критерий случайно сослался. В соседний надо было ) Проблема в отсутствии `<form>` вокруг селекта.
https://up.htmlacademy.ru/htmlcss/33/criteries#HTML-04
Исключение - селект для сортировки. Этот селект - не для сортировки. Поэтому форма вокруг него должна быть.

fesspav replied:
![image](https://user-images.githubusercontent.com/48926104/151351650-5c34fd87-41eb-459e-aa82-5f7ef3b51930.png)
типа такого?

fesspav replied:
только label тоже нужно в form обернуть

efiand replied:
Да, если form вынуть наружу лейбла, то получится то что надо. Как-то принято для фильтров и сортировки метод get, но и через post можно реализовать, в принципе.
Только я как не любитель лишних оберток сделал бы form class="count-pages" без дива. Если во враппер вложен один дочерний элемент, и при этом не решается задача центровки (вертикальной или горизонтальной), то 99%, что этот враппер не нужен.
Исключение - если у нас полный набор макетов, и мы знаем, что тут что-то может быть после формы. Но у нас элемент один. Если когда-нибудь будет не один, то все равно в это место в разметку полезем, тогда и враппер можно накинуть. Их лучше по мере требования делать.

efiand commented on source/catalog.html:
```diff
@@ -0,0 +1,207 @@
+    <div class="count-pages">
+      <label>Гостиниц на странице:
+        <select name="housing-count">
+          <option value="current">4</option>
```
Странное значение для числа "4" :-)

fesspav replied:
типа "текущее количество"

efiand replied:
Но ведь текущее же будет каждый раз разное - в зависимости от того, что выберет пользователь. На макете это 4, с учетом кратности сетки трем заказчик скорее всего попросит что-то типа 9, плюс юзер может выбрать остальные варианты, и они все будут текущими. Делается иначе - через атрибут `selected` (тренажеры для себя освежать идея хорошая):

```html
<select name="limit">
  <option selected>9</option>
  <option>12</option>
  <option>15</option>
  <option>18</option>
</select>
```

efiand commented on source/catalog.html:
```diff
@@ -0,0 +1,207 @@
+    </div>
+  </section>
+
+  <div class="services">
```
Сборка позволяет не дублировать такие штуки - либо вынести в footer, либо вынести за блок main - никто ведь не говорит, что тег `<main>` обязательно тащить в `block main`
Отличие в том, что на главной доп. класс, дающий другой фон и белый цвет тексту

fesspav committed:
внес правки в index

внес правки в index согласно комментариев.

fesspav committed:
обновил каталог согласно комментариям

в GHD в левом меню выбрал нужный файл каталога и запушил.

fesspav committed:
пушу "верхние" файлы сборки, в которых происходит вся работа.

обновил названия некоторых классов, селекторов, добавил див для удобства обращения к элементам списков.
